enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,599 @@
+
+The cluster admin can configure a Gateway with the following three steps:
+
+1. Create a GatewayClass with `spec.controllerName: openshift.io/gateway-controller`.
+2. Create a secret with a default certificate.
+3. Create a Gateway with listeners for HTTP and HTTPS, using the secret from Step 2 and the GatewayClass from Step 1.
+
+The Ingress Operator and OpenShift Service Mesh Operator handle all the details
+of configuring Envoy, provisioning a load balancer, and configuring DNS,
+providing a turnkey ingress solution that immediately enables project admins to
+publish their applications using the Gateway that the cluster admin has created.
+
+For example, the cluster admin can use the following GatewayClass manifest:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: openshift-default
+spec:
+  controllerName: openshift.io/gateway-controller
+```
+
+Then the cluster admin must create a secret with the default certificate:
+
+```shell
+oc -n openshift-ingress create secret tls gwapi-wildcard --cert=wildcard.crt --key=wildcard.key
+```
+
+Finally, the cluster admin can use a Gateway manifest like the following:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: openshift-ingress
+spec:
+  selector:                                                         # (1)
+    istio: ingressgateway
+  gatewayClassName: istio                                           # (2)
+  addresses:
+  - value: istio-ingressgateway.openshift-ingress.svc.cluster.local # (3)
+    type: Hostname
+  listeners:
+  - name: http
+    hostname: "*.gwapi.example.com"
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: https
+    hostname: "*.gwapi.example.com"
+    port: 443
+    protocol: HTTPS
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - name: gwapi-wildcard
+    allowedRoutes:
+      namespaces:
+        from: All
+```
+[1]: The requirement to specify `selector.istio` is an Istio-ism that needs to be investigated. \
+[2]: The requirement to specify `gatewayClassName: istio ` is another Istio-ism that is addressed by \
+     [istio/istio#38134](https://github.com/istio/istio/issues/38134) and \
+     [istio/istio#41884](https://github.com/istio/istio/pull/41884). \
+[3]: The requirement to specify the "istio-ingressgateway" service's DNS name \
+     in `spec.addresses` is another Istio-ism that needs to be discussed.
+
+Then a project admin can attach an HTTPRoute to the Gateway.
+
+_"As a cluster admin, I want to configure ingress using Gateway API, and I want
+to enable application owners to configure mesh using the Istio API, and I want
+these two things to work harmoniously."_
+
+The goals of this enhancement include that ingress and mesh use a unified
+control-plane (with a single Istiod instance).  The cluster admin should be able
```

Comment:
This is the first mention of a unified control-plane (I try to expand initialisms and define OSSM-specific terms the first time I use them within each section).  Maybe it would be better to avoid using the term "Istiod".  Is "Istio control-plane deployment" good?